src/js/utils/CRONValidatorUtil.js

```diff
@@ -0,0 +1,214 @@
+
+/**
+ * Validator utilities for CRON syntax
+ */
+class CRONValidatorUtil {
+
+  /**
+   * Test if the given string is a valid CRON syntax
+   *
+   * @param {string} value - The cron string to test
+   * @returns {string} - Returns null if test passes otherwise return the error string
+   */
+  static testCronString(value) {
+    let test;
+    let parts = value.split(/ +/);
+    if (parts.length !== 5) {
+      return 'Expecting 5 components: minute hour date month weekday';
+    }
+
+    // Validate minute
+    test = CRONValidatorUtil.testCronComponent( parts[0], 0, 59 );
+    if (test !== null) {
+      return 'Minute (1rst component) '+test;
+    }
+
+    // Validate hour
+    test = CRONValidatorUtil.testCronComponent( parts[1], 0, 23 );
+    if (test !== null) {
+      return 'Hour (2nd component) '+test;
+    }
+
+    // Validate date
+    test = CRONValidatorUtil.testCronComponent( parts[2], 1, 31 );
+    if (test !== null) {
+      return 'Date (3rd component) '+test;
+    }
+
+    // Validate month
+    test = CRONValidatorUtil.testCronComponent( parts[3], 1, 12,
+      ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec']
+    );
+    if (test !== null) {
+      return 'Month (4th component) '+test;
+    }
+
+    // Validate weekday
+    test = CRONValidatorUtil.testCronComponent( parts[4], 0, 6,
+      ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun']
+    );
+    if (test !== null) {
+      return 'Week day (5th component) '+test;
+    }
+
+    // All tests passed successfully
+    return null;
+  }
+
+  /**
+   * Check if the given string is a valid cron component
+   *
+   * @param {string} component - The cron component to test
+   * @param {number} numMin - The minimum value accepted in the numerical components
+   * @param {number} numMax - The maximum value accepted in the numerical components
+   * @param {array} discreteValues - Other valid discreet values
+   * @returns {string} - Returns null if test passes otherwise return the error string
+   */
+  static testCronComponent(component, numMin, numMax, discreteValues = []) {
```

Comment:
Any chance that can refactor this, so that he util only returns a boolean? Or at least doesn't need to know about the error messages, maybe by introducing "exit" codes? Nonetheless we should strive to make this util consistent with the other validator utils.